File: Backend/API/Shared/Entities/Booking.cs
using Shared.Enums;

namespace Shared.Entities
{
    public class Booking : BaseEntity
    {
        public string UserEmail { get; set; }

        public DateTime PickupDate { get; set; }

        public DateTime DropoffDate { get; set; }

        public PickupDropoffLocation PickupLocation { get; set; }

        public int PickupLocationId { get; set; }   

        public PickupDropoffLocation DropoffLocation { get; set; }

        public int DropoffLocationId { get; set; }

        public bool IsGuest { get; set; }

        public bool IsApproved { get; set; }

        public int VehicleInsuranceDetailId { get; set; }

        public VehicleInsuranceDetails VehicleInsuranceDetail { get; set; }

        public bool ChildSeatOption { get; set; }

        public PaymentMethodEnum PaymentMethod { get; set; }

        public bool MarketingCheckbox { get; set; }

        public string PromoCode { get; set; }

    }
}

File: Frontend/Admin/src/app/shared/models/booking.ts
export class Booking {
  id: string;
  vehicleId: number;
  userEmail: string;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  pickupDate: Date;
  dropoffDate: Date;
  pickupLocationId: number;
  dropoffLocationId: number;
  isGuest: boolean;
  isApproved: boolean;
  pickupLocation: string;
  dropoffLocation: string;
  userFullName: string;
}
